Week5/app/src/main/java/com/em4n0101/mymovies/profile/ProfileFragment.kt

```diff
@@ -0,0 +1,91 @@
+package com.em4n0101.mymovies.profile
+
+import android.content.Intent
+import android.net.Uri
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.View
+import com.em4n0101.mymovies.R
+import com.em4n0101.mymovies.login.LoginActivity
+import com.em4n0101.mymovies.repositories.SharedPrefsRepository
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import kotlinx.android.synthetic.main.fragment_profile.*
+
+class ProfileFragment : Fragment(R.layout.fragment_profile) {
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        textViewUsername.text = SharedPrefsRepository.getUserName()
+        buttonSignOut.setOnClickListener {
+            navigateToSignIn()
+        }
+        buttonAbout.setOnClickListener {
+            showAbout()
+        }
+        buttonChangeProfileImage.setOnClickListener {
+            chooseProfileImage()
+        }
+
+        activity?.title = getString(R.string.profile)
+
+        // Get profile image
+        val profileImage = SharedPrefsRepository.getProfileImage()
+        if (profileImage != null && profileImage.isNotEmpty()) {
+            imageViewProfile.setImageURI(Uri.parse(profileImage))
+        }
+    }
+
+    private fun chooseProfileImage() {
+        val intent = Intent()
+        intent.type = "image/*"
+        intent.action = Intent.ACTION_OPEN_DOCUMENT
+        intent.addCategory(Intent.CATEGORY_OPENABLE)
+
+        startActivityForResult(Intent.createChooser(intent, getString(R.string.select_picture)), REQUEST_PROFILE_IMAGE)
+    }
+
+    private fun showAbout() {
+        activity?.let {
+            MaterialAlertDialogBuilder(it)
+                .setTitle(resources.getString(R.string.about_title))
+                .setMessage(resources.getString(R.string.about_message))
+                .show()
+        }
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+
+        if (requestCode == REQUEST_PROFILE_IMAGE) {
```

Comment:
you should check here if the `resultCode == Activity.RESULT_OK` as well because it can fail and then you would try to store an image which doesn't exist